pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -618,31 +621,100 @@ func (c *Cluster) saveMonConfig() error {
 var updateDeploymentAndWait = UpdateCephDeploymentAndWait
 
 func (c *Cluster) startMon(m *monConfig, hostname string) error {
-
+	// check if the monitor deployment already exists
 	d := c.makeDeployment(m, hostname)
-	logger.Debugf("Starting mon: %+v", d.Name)
-	_, err := c.context.Clientset.AppsV1().Deployments(c.Namespace).Create(d)
+	var deploymentExists bool
```

Comment:
nit: I'd suggest to set the default to `false`, then only set to true when it's found. It will simplify the code path so you don't need to set it to false later. This pattern will simplify other places such as `pvcExists`.